apache2.yaml

```diff
@@ -158,71 +158,53 @@ subpackages:
 
+          mkdir -p "${{targets.subpkgdir}}"/usr/local/apache2/
+          mkdir -p "${{targets.subpkgdir}}"/usr/local/apache2/conf
+          mkdir -p "${{targets.subpkgdir}}"/usr/local/apache2/var/run/apache2
+
+          mkdir -p "${{targets.subpkgdir}}"/etc/
+          install -m644 -D httpd.conf "${{targets.subpkgdir}}"/etc/
+
+          # Create symlinks
+          ln -s /etc/httpd.conf "${{targets.subpkgdir}}"/usr/local/apache2/conf/
+          ln -s /etc/extra "${{targets.subpkgdir}}"/usr/local/apache2/conf/
+          ln -s /etc/mime.types "${{targets.subpkgdir}}"/usr/local/apache2/conf/
+          ln -s /etc/magic "${{targets.subpkgdir}}"/usr/local/apache2/conf/
+          ln -s /usr/lib/apache2/modules/ "${{targets.subpkgdir}}"/usr/local/apache2/
+          ln -s /usr/share/apache2/default-site/htdocs "${{targets.subpkgdir}}"/usr/local/apache2/
+          ln -s /usr/lib/cgi-bin/ "${{targets.subpkgdir}}"/usr/local/apache2/
+
 test:
   environment:
     contents:
       packages:
         - curl
+        - ${{package.name}}-compat
   pipeline:
     - runs: |
         adduser -D -H www-data
 
         httpd -M
 
-        mkdir -p /usr/local/apache2/conf
-        mkdir -p /usr/local/apache2/htdocs
-        mkdir -p /usr/local/apache2/logs
-        mkdir -p /usr/local/apache2/var/run/apache2
-
-        # Set the document root path based on the system
-        DOC_ROOT="/usr/local/apache2/htdocs"
-        CONF_FILE="/usr/local/apache2/conf/httpd.conf"
-
-        # Add the required configuration to allow access for all
-        echo "Modifying Apache configuration to allow access to all..."
-        cat <<EOL >> "$CONF_FILE"
-        ServerRoot "/usr/local/apache2"
-        Listen 80
-        ServerName localhost:8080
-
-        # Load required modules
-        LoadModule mpm_prefork_module /usr/lib/apache2/modules/mod_mpm_prefork.so
-        LoadModule authz_core_module /usr/lib/apache2/modules/mod_authz_core.so
-        LoadModule log_config_module /usr/lib/apache2/modules/mod_log_config.so
-        LoadModule unixd_module /usr/lib/apache2/modules/mod_unixd.so
-
-        DocumentRoot "/usr/local/apache2/htdocs"
-
-        # Custom configuration to allow access to all
-        <Directory "$DOC_ROOT">
-            Options Indexes FollowSymLinks
-            AllowOverride None
-            Require all granted
-        </Directory>
-
-        EOL
-
-        # Ensure correct permissions on the document root
-        echo "Setting permissions for the document root..."
-        chmod o+rx /usr/local/apache2 "$DOC_ROOT"
-        chown -R $(whoami):$(whoami) "$DOC_ROOT"
-        chmod -R 755 "$DOC_ROOT"
-
+        # Add page to show
         echo "<html><body><h1>Apache Test Page</h1></body></html>" > /usr/local/apache2/htdocs/test.html
 
 
         # Start httpd using the custom configuration
         echo "Starting httpd..."
-        httpd -f "$CONF_FILE" -DFOREGROUND &
+        httpd -DFOREGROUND &
```

Comment:
Two things here and then I think it is ready to ship.

1. Ideally there would be a test step before this one that uses the default /usr/share/apache2 and /etc/apache2 directories, and the default httpd.conf file that ships with the `make install` step. That way running `apk install apache2` results in a working server.

2. For this compat package, could it use `sed` on a copy of the default package's httpd.conf file to update the `/usr/local/apache2` paths and Error/Custom log directives? That way there's no need to vendor a copy in case something changes in Apache's sources. Then the test pipeline can still use it, without clobbering the working main package.

Reply:
@jamonation I believe the current changes should now address your suggestions.